djstripe/contrib/rest_framework/mixins.py

```diff
@@ -0,0 +1,43 @@
+from ...models import Customer
+from ...settings import STRIPE_LIVE_MODE, subscriber_request_callback
+
+
+class AutoCreateCustomerMixin:
+    """Small mixin that can be included in REST API Views.
+
+    If included, it will automatically create a Customer instance
+    for the authenticated user, if does not yet exist.
+    """
+
+    def dispatch(self, request, *args, **kwargs):
+        """Automatically creates a Customer if it does not exists yet.
+
+        The dispatch method is called after permissions and auth have been resolved,
+        but before the actual get/create/update methods are called.
+        """
+        result = super().dispatch(request, *args, **kwargs)
+        if not request.user.is_anonymous:
+            Customer.objects.get_or_create(
```

Comment:
This doesnt support the new 'account' field added to Customer.

IMO this `AutoCreateCustomerMixin` should only be used on the old deprecated end-points.

I suspect the current logic is assuming that dj-stripe is not kept in sync, so creating a Customer is a hack to simplify fetching a Subscription from Stripe to add to the local dj-stripe dataset.

IMO the new subscription end-points should not use any Customer `create` methods.  `get`s all the way.  If no subscriptions could be found, no customer records should be created by a read view. (the subscriber record will also be put into third party Stripe, all because of a 'read' of an 'empty' list - data policy people cry about this stuff)

Also it is possible for one Subscriber to be connected to multiple 'account' in Stripe.  At the moment that is represented as multiple Customer objects for the same Subscriber. (not ideal, but it works).
Assuming the dj-stripe data is in sync, iterating over the existing Customer for the Subscriber is sufficient.

Reply:
I agree this `AutoCreateCustomerMixin ` is a hack and should be removed.